lib/core_to_mucore.ml

```diff
@@ -1091,13 +1146,22 @@ let make_fun_with_spec_args f_i loc env args (accesses, requires) =
       (*   let here = Locations.other __LOC__ in *)
       (*   (LTranslate.T (IT.good_ (ct, IT.sym_ (pure_arg, bt, here)) here), info) *)
       (* in *)
-      let@ at = aux (* good_lc :: *) good_lcs env st rest in
+      let@ at = aux_comp (* good_lc :: *) good_lcs env st rest in
       return (Mu.mComputational ((pure_arg, bt), (loc, None)) at)
     | [] ->
       let@ lat = make_largs_with_accesses f_i env st (accesses, requires) in
       return (Mu.L (Mu.mConstraints (List.rev good_lcs) lat))
   in
-  aux [] env Translate.C_vars.init args
+  let rec aux_ghost env st = function
+    | (arg, cnbt) :: rest ->
+      let sbt = Translate.base_type env cnbt in
+      let bt = SBT.proj sbt in
+      let env = Translate.add_logical arg sbt env in
+      let@ at = aux_ghost env st rest in
+      return (Mu.Ghost ((arg, bt), (loc, None), at))
+    | [] -> aux_comp [] env st args
```

Comment:
Isn't there convention/code that relies on computational arguments being first?

Reply:
Not that I'm aware of, but I refactored so computational arguments go first.